### PR TITLE
C4 audit mitigations

### PR DESCRIPTION
**Pull request by PierrickGT**
This PR fixes the following issues:
- https://github.com/code-423n4/2021-06-pooltogether-findings/issues/95
- https://github.com/code-423n4/2021-06-pooltogether-findings/issues/90
- https://github.com/code-423n4/2021-06-pooltogether-findings/issues/71
- https://github.com/code-423n4/2021-06-pooltogether-findings/issues/60
- https://github.com/code-423n4/2021-06-pooltogether-findings/issues/46
- https://github.com/code-423n4/2021-06-pooltogether-findings/issues/44
- https://github.com/code-423n4/2021-06-pooltogether-findings/issues/43
- https://github.com/code-423n4/2021-06-pooltogether-findings/issues/42
- https://github.com/code-423n4/2021-06-pooltogether-findings/issues/2

### CODE REVIEW

File: contracts/yield-source/YearnV2YieldSource.sol [hunk 1]
```diff
@@ -261,7 +313,7 @@ contract YearnV2YieldSource is IYieldSource, ERC20Upgradeable, OwnableUpgradeabl
     /// @dev used to calculate how many shares to mint / burn when depositing / withdrawing
     /// @return tokens number of tokens equivalent (in value) to the amount of Yield Source shares
     function _sharesToToken(uint256 shares) internal view returns (uint256 tokens) {
-        if(totalSupply() == 0) {
+        if (totalSupply() == 0) {
```
Comment (asselstine) [inline contracts/yield-source/YearnV2YieldSource.sol]: let's pull this into a variable, as it's a public function call

File: contracts/yield-source/YearnV2YieldSource.sol [hunk 2]
```diff
@@ -248,7 +300,7 @@ contract YearnV2YieldSource is IYieldSource, ERC20Upgradeable, OwnableUpgradeabl
     /// @param tokens amount of tokens to be converted
     /// @return shares number of shares equivalent to the amount of tokens
     function _tokenToShares(uint256 tokens) internal view returns (uint256 shares) {
-        if(totalSupply() == 0) {
+        if (totalSupply() == 0) {
```
Comment (asselstine) [inline contracts/yield-source/YearnV2YieldSource.sol]: let's pull this into a variable

**Review by asselstine — CHANGES_REQUESTED** [review]

File: contracts/yield-source/YearnV2YieldSource.sol (hunk 1 above)
Comment (PierrickGT) [reply contracts/yield-source/YearnV2YieldSource.sol]: Done: https://github.com/pooltogether/pooltogether-yearnv2-yield-source/pull/9/commits/32b130dbbaf28b614075ce2d30b0ee7c9f47593c

File: contracts/yield-source/YearnV2YieldSource.sol (hunk 2 above)
Comment (PierrickGT) [reply contracts/yield-source/YearnV2YieldSource.sol]: Done: https://github.com/pooltogether/pooltogether-yearnv2-yield-source/pull/9/commits/32b130dbbaf28b614075ce2d30b0ee7c9f47593c

**Review by asselstine — APPROVED** [review]

**Review by asselstine — APPROVED** [review]

**Pull request merged**